### PR TITLE
Rename CloudBaseHook to GoogleBaseHook and move it to google.common

### PR DESCRIPTION
**Pull request by turbaszek**
---
Issue link: WILL BE INSERTED BY [boring-cyborg](https://github.com/kaxil/boring-cyborg)

Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

File: airflow/providers/google/cloud/hooks/discovery_api.py [hunk 1]
```diff
@@ -23,10 +23,10 @@
 
 from googleapiclient.discovery import Resource, build
 
-from airflow.providers.google.cloud.hooks.base import CloudBaseHook
+from airflow.providers.google.common.hooks.base import GoogleBaseHook
```
Comment (mik-laj) [inline airflow/providers/google/cloud/hooks/discovery_api.py]: Should we move this hook to common package?

**Review by mik-laj — APPROVED** [review]

File: airflow/providers/google/cloud/hooks/discovery_api.py (hunk 1 above)
Comment (turbaszek) [reply airflow/providers/google/cloud/hooks/discovery_api.py]: I think so, good catch! @potiuk WDYT?

**Review by potiuk — APPROVED** [review]

**Comment by mik-laj** [comment]
Hi. Travis is sad. This is a blocker for me because I would like to take care of the latest improvements. in the package/class name.

File: airflow/providers/google/cloud/hooks/discovery_api.py (hunk 1 above)
Comment (mik-laj) [reply airflow/providers/google/cloud/hooks/discovery_api.py]: I will do it in a separate PR.

Comment (mik-laj) [reply airflow/providers/google/cloud/hooks/discovery_api.py]: Thanks for doing this.

File: airflow/providers/google/cloud/hooks/automl.py [hunk 2]
```diff
@@ -29,10 +29,10 @@
     ImageObjectDetectionModelDeploymentMetadata, InputConfig, Model, Operation, PredictResponse, TableSpec,
 )
 
-from airflow.providers.google.cloud.hooks.base import CloudBaseHook
+from airflow.providers.google.common.hooks.base import GoogleBaseHook
```
Comment (mik-laj) [inline airflow/providers/google/cloud/hooks/automl.py]:
```suggestion
from airflow.providers.google.common.hooks.base_google import GoogleBaseHook
```
We agreed to such a convention in the case of AWS. In the case of Google, we should also use it.  I will add a commit to this branch, which corrects it. In next step,  I will start new PR starting from this branch, which will correct other errors.

**Pull request merged**